language/move-vm/runtime/src/loader.rs

```diff
@@ -537,7 +537,9 @@ impl Loader {
         ty_args: &[TypeTag],
         data_store: &mut impl DataStore,
     ) -> VMResult<(Arc<Function>, Vec<Type>)> {
-        self.load_module_expect_no_missing_dependencies(module_id, data_store)?;
+        let mut visited_deps = HashSet::new();
+        visited_deps.insert(module_id.clone());
+        self.load_module(module_id, data_store, &mut visited_deps, false)?;
```

Comment:
I don't understand what moving `visited_deps` to `self` would do, besides the fact I would never do that because it would be promoting something that must live on the stack to global scope.
We would also have to have an API in the caller to clean up the `visited_deps` which amounts to the same problem.
I think this is a caller responsibility and it should be clearly in your face